app/models/cohort_columns/last_homeless_visit.rb

```diff
@@ -27,9 +27,16 @@ def value(cohort_client, user) # OK
       # e.g.: {:project_name=>\"APR - Transitional Housing\", :date=>Mon, 30 Sep 2019, :project_id=>10}
       return unless lhv.present?
 
+      enforce_visibility = cohort_client.cohort.enforce_project_visibility_on_cells?
       lhv = JSON.parse(lhv)
       lhv.select do |row|
-        row['project_id'].in? user.visible_project_ids_enrollment_context
+        next row['project_id'].in?(user.visible_project_ids_enrollment_context) if enforce_visibility
+        next true unless cohort.only_window?
+
+        window_project_ids = GrdaWarehouse::Hud::Project.joins(:data_source).
```

Comment:
Promote this up out of the loop to avoid repeated queries?

Reply:
Done.